### ISSUE TITLE
Disabling SC1072,SC1073 does not allow further checks

### ISSUE THREAD

**Issue by harleypig** [koalaman/shellcheck#1291]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1073, SC1064, SC1072

SC1072,SC1073

- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#1150 is probably related: "[...] as a current design decision, ShellCheck only accepts { .. } and ( .. ) for function definitions."

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# shellcheck disable=SC1072,SC1073,SC1064
function is_integer() [[ "$1" != *[!0-9]* ]]

is_integer 32 && echo 'yep'
```

#### Here's what shellcheck currently says:
```
Line 4	SC1073: Couldn't parse this function. Fix to allow more checks.
Line 4	SC1072:  Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

Nothing, or further errors and warnings if any. I disabled the related codes, shellcheck should not hang here.

**Comment by koalaman** [comment]
Sorry, silencing fatal parser errors is purely cosmetic; any parser error found will still stop ShellCheck at the point of the error. There's no great way of resuming parsing of unrecognized code because you don't know what kind of state you'd be in.

**Comment by matthewpersico** [comment]
Why would you even want to silence fatal errors?

**Comment by koalaman** [comment]
ShellCheck doesn't always make it clear that the error is fatal and has prevented further processing. It's only recently that it's started saying "Fix to enable more checks".

**Comment by harleypig** [comment]
@matthewpersico: I didn't realize it was a fatal error when I submitted the issue. I don't think it should be a fatal error because it's a valid bash function definition, but I understand design decisions. And it's a very useful app as it is, so I can live with it.
